akka-discovery-azure-api/src/main/scala/akka/discovery/azureapi/JsonFormat.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2017-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.discovery.azureapi
+
+import akka.annotation.InternalApi
+import akka.discovery.azureapi.PodList._
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import spray.json._
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[azureapi] object JsonFormat extends SprayJsonSupport with DefaultJsonProtocol {
+  // If adding more formats here, remember to also add in META-INF/native-image reflect config
```

Comment:
Not entirely sure we need to support native image with this module, but if we should, there should be a `META-INF/native-image/com.lightbend.akka.management.akka-discovery-azure-api/reflect-config.json` listing the model classes. You can look at the corresponding `reflect-config.json` in lease-kubernetes for an example.

`jsonFormatN` inspects them using reflection so must explicitly be made available for native image/graal through such config.

Reply:
I plan on doing native image later in a different PR.